metrics/tags.go

```diff
@@ -3,70 +3,143 @@ package metrics
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"sort"
 	"strings"
 
+	"github.com/mailru/easyjson"
+	"github.com/mailru/easyjson/jlexer"
+	"github.com/mailru/easyjson/jwriter"
 	"github.com/mstoykov/atlas"
 )
 
-// TagSet represents a set of tags.
-type TagSet struct {
-	tags *atlas.Node
+// A TagSet represents an immutable set of metric tags. For the efficient and
+// thread-safe storage of the key=value tag pairs, it uses the
+// https://github.com/mstoykov/atlas data structure.
+type TagSet atlas.Node
+
+// With returns another TagSet object that contains the combination of the
+// current receiver tags and the name=value tag from its parameters. It doesn't
+// modify the receiver, it will either return an already existing TagSet with
+// these tags, if it exists, or create a new TagSet with them and return it.
+func (ts *TagSet) With(name, value string) *TagSet {
+	return (*TagSet)(((*atlas.Node)(ts)).AddLink(name, value))
 }
 
-// TagSetFromSampleTags creates a TagSet starting
-// from the set defined by SampleTags.
-// Adding a tag on the new set doesn't impact the SampleTags.
-func TagSetFromSampleTags(st *SampleTags) *TagSet {
-	tm := &TagSet{}
-	tm.tags = st.tags
-	return tm
+// Without returns another TagSet, either an already existing one or a newly
```

Comment:
While I really like both names, I feel like  they are so close that I will personally misread `Without` and `With` when going over code quickly. Does anyone else thinks this is a problem? Arguably it should be fairly obvious which one was intended to use, and so unlikely that you will actually not know what happens. 

nit: Also you can word both documentaiton in similar way :)

Reply:
:+1: for making the comments similar, :-1: for changing the names. They are dissimilar enough, and they accept a different number of parameters, so I don't think anyone will confuse them

Reply:
Resolved by https://github.com/grafana/k6/pull/2638/commits/eec1bc9afb712ba699034f6d27a5eacb56e21017